### PR TITLE
User-assigned Managed Identity for VMs

### PR DESCRIPTION
**Pull request by Garth-CD**
Added additional User-assigned Managed Identity for AMA deployment Policy, to be deployed along with other bootstrap managed identities.

This additional identity is required when setting up the AMA deployment initiative.

### CODE REVIEW

**Comment by richeney** [comment]
@Garth-CD there is already a second managed identity for use with Azure Policy, so why do we need a third? The PR comment above does not explain it effectively. 

Note that Azure Policy's Deploy If Not Exists effect is configured with an array of roleDefinitionIds. You need a privileged role for policy assignments because it is capable of assigning roles on the fly when remediating. By default the policy managed identity has no RBAC permissions.

Your additional vm management identity repeats the comment for the policy identity and it is unclear why it is needed especially as it has no RBAC role assignments. Remember this repo is public so the commenting needs to be full and accurate.

**Comment by Garth-CD** [comment]
@richeney The AMA agent requires a User or System Assigned identity in order to function. We make use of the User Assigned identity policies as that allows mapping one single identity to multiple workloads instead of having each VM with a System Assigned identity which clogs up Entra ID with all of these Managed Identities
 
The AMA User identity policies need an identity to remediate resources. This is the `id-cdmonitoring-policy-prod-uksouth-001` identity which is currently deployed via the Terraform Bootstrap.
 
The new proposed identity would be the identity that is assigned to VMs themselves which has no permissions. There are 2 problems this new VM identity would solve:
 

- The policy initiatives have the option of creating and assigning a built-in User Identity which is created by the policy. If we use the default built-in identity that the policy can assign that might not work due to other deny policies in the target environment which are enforcing tags. naming conventions etc

- If we use the policy identity for both the policy itself as well as the VMs then that means the VMs have an identity assigned which has a whole bunch of potentially privileged permissions which we don't want

File: main.tf [hunk 1]
```diff
@@ -90,6 +90,13 @@ resource "azurerm_user_assigned_identity" "policy" {
   location            = azurerm_resource_group.rg.location
 }
 
+// For optional use in policy assignments. Better naming convention than the system generated name.
```
Comment (richeney) [inline main.tf:93]: @Garth-CD please update this comment for clarity

**Comment by Garth-CD** [comment]
@richeney Done.

**Review by richeney — APPROVED** [review]
lgtm - thanks for the additional detail.

One last request. We have an FAQ that we want to make publicly available. I suggest we add a little detail before doing so?

<https://github.com/Cloud-Direct-Monitoring/.github-private/blob/main/FAQ.md>

**Pull request merged**